### PR TITLE
fixed openid username

### LINKED ISSUE #1747: Set a more meaningful openid username
When creating a new account with openid, the username depends on whether or not the service is returning a username field (auth.info.username), if not the username becomes like 1234@facebook (uuid@provider), which does not mean anything to users reading it, and conveys no information whatsoever regarding who this user is. 

We should setup the username during user creation in a more meaningful way, such as: 
If the email is user_stella@gmail.com, the username should become user_stella@facebook.

### PR DESCRIPTION
**Pull request by ViditChitkara**
closes #1747

### CODE REVIEW

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -11,13 +11,13 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
     def handle(provider)
       auth_hash = request.env['omniauth.auth']
-      uid = auth_hash[:uid]
+      username = auth_hash.info.email.split("@")[0]
```
Comment (differentreality) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Use single quotes here, it's a styling thing, and the reason why linters have failed

Comment (ViditChitkara) [reply app/controllers/users/omniauth_callbacks_controller.rb]: Sure @differentreality

**Comment by differentreality** [comment]
:+1: Nice @ViditChitkara

**Comment by ViditChitkara** [comment]
ok

**Comment by differentreality** [comment]
Thank you @ViditChitkara :tada:

**Pull request merged**